### ISSUE TITLE
Windows TaskCluster runs fail on error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version

### ISSUE THREAD

**Issue by mykmelez** [mozilla/qbrt#142]
The Windows TaskCluster runs are failing on `error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version` when cloning the repo from GitHub, for example [on this run](https://tools.taskcluster.net/groups/O1NtUhUpQrS0WZpsIhYUrw/tasks/O1NtUhUpQrS0WZpsIhYUrw/runs/0/):

```
[taskcluster 2018-03-08T01:32:57.293Z] Worker Type (win2012r2) settings:
[taskcluster 2018-03-08T01:32:57.293Z]   {
[taskcluster 2018-03-08T01:32:57.293Z]     "aws": {
[taskcluster 2018-03-08T01:32:57.293Z]       "ami-id": "ami-9fb400e7",
[taskcluster 2018-03-08T01:32:57.293Z]       "availability-zone": "us-west-2c",
[taskcluster 2018-03-08T01:32:57.293Z]       "instance-id": "i-0f7d1ab1505da33e3",
[taskcluster 2018-03-08T01:32:57.293Z]       "instance-type": "m3.2xlarge",
[taskcluster 2018-03-08T01:32:57.293Z]       "local-ipv4": "172.31.5.183",
[taskcluster 2018-03-08T01:32:57.293Z]       "public-hostname": "ec2-54-202-165-106.us-west-2.compute.amazonaws.com",
[taskcluster 2018-03-08T01:32:57.293Z]       "public-ipv4": "54.202.165.106"
[taskcluster 2018-03-08T01:32:57.293Z]     },
[taskcluster 2018-03-08T01:32:57.293Z]     "config": {
[taskcluster 2018-03-08T01:32:57.293Z]       "deploymentId": "b_-9AJBeQ6CPhqsSxu-VeQ",
[taskcluster 2018-03-08T01:32:57.293Z]       "runTasksAsCurrentUser": false
[taskcluster 2018-03-08T01:32:57.293Z]     },
[taskcluster 2018-03-08T01:32:57.293Z]     "generic-worker": {
[taskcluster 2018-03-08T01:32:57.293Z]       "go-arch": "amd64",
[taskcluster 2018-03-08T01:32:57.293Z]       "go-os": "windows",
[taskcluster 2018-03-08T01:32:57.293Z]       "go-version": "go1.9",
[taskcluster 2018-03-08T01:32:57.293Z]       "release": "https://github.com/taskcluster/generic-worker/releases/tag/v10.4.1",
[taskcluster 2018-03-08T01:32:57.293Z]       "revision": "c0f768e3d1017032aa417400215ff5f422ab686f",
[taskcluster 2018-03-08T01:32:57.293Z]       "source": "https://github.com/taskcluster/generic-worker/commits/c0f768e3d1017032aa417400215ff5f422ab686f",
[taskcluster 2018-03-08T01:32:57.293Z]       "version": "10.4.1"
[taskcluster 2018-03-08T01:32:57.293Z]     },
[taskcluster 2018-03-08T01:32:57.293Z]     "machine-setup": {
[taskcluster 2018-03-08T01:32:57.293Z]       "maintainer": "pmoore@mozilla.com",
[taskcluster 2018-03-08T01:32:57.293Z]       "script": "https://raw.githubusercontent.com/taskcluster/generic-worker/6ff29c340888215bd3c212e7ecfb66dffe95c18b/worker_types/win2012r2/userdata"
[taskcluster 2018-03-08T01:32:57.293Z]     }
[taskcluster 2018-03-08T01:32:57.293Z]   }
[taskcluster 2018-03-08T01:32:57.293Z] Task ID: O1NtUhUpQrS0WZpsIhYUrw
[taskcluster 2018-03-08T01:32:57.293Z] === Task Starting ===
[taskcluster 2018-03-08T01:33:00.520Z] Uploading redirect artifact public/logs/live.log to URL https://g3fkk2qaaaawea3uki3wlumamjxt6hatdcjsro54afj7uzib.taskcluster-worker.net:60023/log/MVRH32cxQ3Oai0OwV2zZDA with mime type "text/plain; charset=utf-8" and expiry 2018-03-08T02:32:57.655Z
[taskcluster 2018-03-08T01:33:01.020Z] Executing command 0: git clone https://github.com/mykmelez/qbrt.git repo

C:\Users\task_1520472698>git clone https://github.com/mykmelez/qbrt.git repo 
Cloning into 'repo'...
fatal: unable to access 'https://github.com/mykmelez/qbrt.git/': error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version
[taskcluster 2018-03-08T01:33:27.076Z]    Exit Code: 128
[taskcluster 2018-03-08T01:33:27.076Z] Success Code: 0x0
[taskcluster 2018-03-08T01:33:27.076Z]    User Time: 0s
[taskcluster 2018-03-08T01:33:27.076Z]  Kernel Time: 0s
[taskcluster 2018-03-08T01:33:27.076Z]    Wall Time: 26.0482038s
[taskcluster 2018-03-08T01:33:27.076Z]  Peak Memory: 1703936
[taskcluster 2018-03-08T01:33:27.076Z]       Result: FAILED
[taskcluster 2018-03-08T01:33:27.077Z] === Task Finished ===
[taskcluster 2018-03-08T01:33:27.077Z] Task Duration: 26.0572061s
[taskcluster 2018-03-08T01:33:27.617Z] Uploading redirect artifact public/logs/live.log to URL https://queue.taskcluster.net/v1/task/O1NtUhUpQrS0WZpsIhYUrw/runs/0/artifacts/public/logs/live_backing.log with mime type "text/plain; charset=utf-8" and expiry 2019-03-09T01:24:53.138Z
[taskcluster 2018-03-08T01:33:27.628Z] Task not successful due to following exception(s):
[taskcluster 2018-03-08T01:33:27.628Z] Exception 1)
[taskcluster 2018-03-08T01:33:27.628Z] []
[taskcluster 2018-03-08T01:33:27.628Z] 
[taskcluster 2018-03-08T01:33:27.628Z] Exit code: 128
[taskcluster 2018-03-08T01:33:27.628Z] 
```

**Comment by mykmelez** [comment]
#143 works around this issue by switching to a custom Git, but I'm going to leave this open to switch back to system Git once https://github.com/taskcluster/generic-worker/pull/76 lands.

**Comment by mykmelez** [comment]
That change has now landed upstream, so #147 undoes the workaround in #143.
